src/roster/views.py

```diff
@@ -1475,6 +1476,15 @@ def __init__(self):
 
     def get(self, request, *args, **kwargs):
         self.clusive_user = request.clusive_user
+
+        if 'days' in kwargs:
+            self.days = kwargs.get('days')
+            logger.debug('Setting student activity days = %d', self.days)
+            self.clusive_user.student_activity_days = self.days
+            self.clusive_user.save()
+        else:
+            self.days = self.clusive_user.student_activity_days
+
         if not self.clusive_user.can_manage_periods:
```

Comment:
I should have noticed this for the time scale PR: I think this check whether the current user can manage periods should happen earlier, right after assigning self.clusive_user. The reason is that there is no need to do anything further if the user cannot navigate to the student details page. If the check fails, self.handle_no_permission() is called and results in a "403 Forbidden" response.

Reply:
Agreed. This check can go right at the top.